Add webapp and hirni

bpoldrack opened the pull request:
- [x] NF: allow to specify github repos outside datalad organization
- [x] NF: allow to specify whether to use `nose` or `pytest` for an extension's tests
- [x] ENH: Add `datalad-webapp` to the to be tested extensions
- [x] ENH: Add `datalad-hirni``to the to be tested extensions

yarikoptic commented on extensions.yaml:
```diff
@@ -28,5 +28,12 @@ extensions:
+  apt_depends: [dcm2niix]
+  pypi: datalad-hirni
+  package: datalad_hirni
+  github: /psychoinformatics-de/datalad-hirni
```
FWIW, I would have kept leading `/` out of this one here, and explicit in the urls like `https://github.com/{e. ...`

bpoldrack replied:
Well, can do it that way, too. Leading `/` is just a slightly simpler thing to check for. Doesn't really matter to me.

yarikoptic commented:
bloody azure ... heh heh

bpoldrack commented:
@yarikoptic : Changed to have that leading `/` and added singualrity as requirement for hirni tests. What's missing is to be able to specify a different testing framework, since webapp is written for pytest not nose. Looking into that the respective templating ...

yarikoptic commented:
It's jinja so I guess you could add a conditional block and introduce a new configuration setting to extensions.yaml .

bpoldrack commented:
Grmpf.

And: hirni test build didn't work, since it failed to run setup.py for `citeproc-py` (which doesn't happen locally for me, of course). Where does that even come from? `duecredit`?

bpoldrack commented:
Ok, now `datalad-hirni` tests are failing, BUT: It's the actual tests! That's fine, I think, since the purpose of this repo and this PR is not to fix datalad or any extension, but to run those tests. They run. 
`datalad-neuroimaging` stalled trying to setup for more than an hour by now. Passed before, nothing changed in that setup - has to be azure or something.

So - I'd vote for merge. Whatever leads to hirni failure is to investigate, but again: The point is to have those tests running not passing.
